### PR TITLE
README: Recommend `configure --config-cache --enable-ccache`

### PR DESCRIPTION
**Pull request by mkoeppe**
This can save a bit of time for developers, although only a fraction of our configure tests are cached.

Also removes mention of  `--enable-download-from-upstream-url`, which has been the default for a while now.

Also reducing some excessive messages from spkg-configure

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -298,12 +298,18 @@ in the Installation Guide.
...
-    A notable option for Sage developers is the following:
-
-    - Use `./configure --enable-download-from-upstream-url` to allow
-      downloading packages from their upstream URL if they cannot (yet) be
-      found on the Sage mirrors. This is useful for trying out ticket branches
-      that make package upgrades.
+    Notable options for Sage developers are the following:
+
+    - Use the option `--config-cache` to have ``configure``
+      keep a disk cache of configuration values. This gives a nice speedup
```
Comment (kwankyu) [inline README.md:325]: Why double backticks?

File: README.md [hunk 2]
```diff
@@ -298,12 +298,18 @@ in the Installation Guide.
...
+      when trying out ticket branches that make package upgrades, which
+      involves automatic re-runs of the configuration step.
+
+    - Use the option `--enable-ccache` to have Sage install and use the
+      optional package ``ccache``, which is preconfigured to keep a
+      disk cache of object files created from source files. This can give
```
Comment (kwankyu) [inline README.md:331]: Why double backticks?

**Comment by kwankyu** [comment]
Otherwise, lgtm.

File: README.md (hunk 1 above)
Comment (mkoeppe) [reply README.md:325]: Thanks, fixed

**Review by kwankyu — APPROVED** [review]
Thanks.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**